features/admin.claims.v1/components/edit/external-dialect/edit-external-claims.tsx

```diff
@@ -159,6 +167,43 @@ export const EditExternalClaims: FunctionComponent<EditExternalClaimsPropsInterf
 
     const eventPublisher: EventPublisher = EventPublisher.getInstance();
 
+    useEffect(() => {
+        if (!attributeUri || !dialectID) {
+            return;
+        }
+        if (!SCIMConfigs.serverSupportedClaimsAvailable.includes(attributeUri)) {
+            return;
+        }
+        getServerSupportedClaimsForSchema(dialectID)
+            .then((response: ServerSupportedClaimsInterface) => {
+                setServerSupportedClaims(response.attributes);
+            })
+            .catch((error: IdentityAppsError) => {
+                dispatch(addAlert({
+                    description: error?.description
+                        || t("claims:local.notifications.getClaims.genericError.description"),
+                    level: AlertLevels.ERROR,
+                    message: error?.message
+                        || t("claims:local.notifications.getClaims.genericError.message")
+                }));
+            });
+    }, [ dialectID ]);
+
+    useEffect(() => {
+        if (!filteredClaims?.length) return;
+        const _extClaims: Set<string> = new Set(filteredClaims.map(
+            (c: ExternalClaim) => c.claimURI
+        ));
+
+        setServerSupportedClaims([
+            ...(serverSupportedClaims ?? []).filter((c: string) => !_extClaims.has(c))
+        ]);
+    }, [ filteredClaims ]);
+
+    useEffect(() => {
+        setHasServerSupportedClaimsToMap(!!serverSupportedClaims?.length);
```

Comment:
Let's do this calculation in a useMemo hook rather than multiple useEffects.

Reply:
Ref;
```
const hasServerSup = useMemo(() => {
	calculate

	return length > 0
}, [serverSupportedClaims, filteredClaims])
```

Reply:
Fixed in 011b0db182c8b70fc2433d301bc44890a41182be